pkg/matrix/matrix.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2022 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package matrix
+
+import (
+	"strconv"
+
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+)
+
+// FanOut produces combinations of Parameters of type String from a slice of Parameters of type Array.
+func FanOut(params []v1beta1.Param) Combinations {
```

Comment:
just a styling comment here, it will be great to see this as a member function of Combination, just like you have other member function at https://github.com/tektoncd/pipeline/pull/4934/files#diff-8ca8f243041fb0ce01d5916da5dcde57488b8a94c7aa294eee17a5b3b9cc5eeaR30

In this case, we are initializing combinations compared to reading from it.

Reply:
good idea, moved them to member functions 👍🏾